SpokestackTests/NLUTensorflowSlotParserTest.swift

```diff
@@ -11,28 +11,60 @@ import Spokestack
         
-        // integer parse
-        let taggedInputRating = zip(["b_rating"],["ten"])
-        let parsedInteger = try! parser.parse(taggedInput: taggedInputRating, intent: metadata.intents.filter({ $0.name == "rate.app" }).first!)
-        XCTAssertEqual(parsedInteger["rating"]!.value as! Int, 10)
+        let taggedInputRating51 = zip(["b_rating", "i_rating", "i_rating", "i_rating"],["fi","##ft", "##ie", "one"])
+        let parsedInteger51 = try! parser.parse(taggedInput: taggedInputRating51, intent: metadata!.intents.filter({ $0.name == "rate.app" }).first!, encoder: encoder!)
+        XCTAssertEqual(parsedInteger51["rating"]!.value as! Int, 51)
+
+        let taggedInputRatingNil = zip(["b_rating", "i_rating", "i_rating", "i_rating"],["fi","##ft", "##ie", "six"])
+        let parsedIntegerNil = try! parser.parse(taggedInput: taggedInputRatingNil, intent: metadata!.intents.filter({ $0.name == "rate.app" }).first!, encoder: encoder!)
+        XCTAssertNil(parsedIntegerNil["rating"]!.value)
+        
+        let taggedInputRatingMillion = zip(["b_iMi", "i_iMi"],["one", "million"])
+        let intent = metadata!.intents.filter({ $0.name == "i.i" }).first!
+        let parsedIntegerMillion = try! parser.parse(taggedInput: taggedInputRatingMillion, intent: intent, encoder: encoder!)
+        XCTAssertEqual(parsedIntegerMillion["iMi"]!.value as! Int, 1000000)
         
-        // digits parse
-        let taggedInputPhone = zip(["b_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number"],["42", "##38", "##34", "##17", "##45"])
-        let parsedDigits = try! parser.parse(taggedInput: taggedInputPhone, intent: metadata.intents.filter({ $0.name == "inform.phone_number" }).first!)
-        XCTAssertEqual(parsedDigits["phone_number"]!.value as! String, "4238341745")
+    }
+    
+    func testParseDigits() {
+        let taggedInputPhoneNumeric = zip(["b_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number"],["42", "##38", "##34", "##17", "##45"])
+        let parsedDigitsNumeric = try! parser.parse(taggedInput: taggedInputPhoneNumeric, intent: metadata!.intents.filter({ $0.name == "inform.phone_number" }).first!, encoder: encoder!)
+        XCTAssertEqual(parsedDigitsNumeric["phone_number"]!.value as! String, "4238341745")
+
+        let taggedInputPhoneCardinal = zip(["b_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number", "i_phone_number"],["4", "second", "three", "eighth", "three", "four", "one", "seven", "four", "five"])
+        let parsedDigitsCardinal = try! parser.parse(taggedInput: taggedInputPhoneCardinal, intent: metadata!.intents.filter({ $0.name == "inform.phone_number" }).first!, encoder: encoder!)
+        XCTAssertEqual(parsedDigitsCardinal["phone_number"]!.value as! String, "4238341745")
+    }
+    
+    func testParseEntity() {
```

Comment:
It might be good to try an input or two with some `"o"` tags and more than one slot value just to exercise those parts of the parsing logic.